### PR TITLE
PullRequest for Issue 4: make the interface configurable

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
Modified:
- rename main.cpp to AMDSMain.cpp

Added:
- argument parser for AMDS application --help for help information, --hw=[hw type] for configuration network hw
- add hw type to the AMDS server thread

https://github.com/acquaman/AcquamanDataServer/issues/4

### CODE REVIEW

File: source/main.cpp [hunk 1]
```diff
@@ -1,12 +0,0 @@
-#include <QtCore/QCoreApplication>
```
Comment (XuetaoLiu-S) [inline source/main.cpp:1]: File main.cpp is renamed as AMDSMain.cpp

**Comment by XuetaoLiu-S** [comment]
@davidChevrier you are so fast. :smiley:

File: source/AMDSThreadedTcpDataServer.h [hunk 2]
```diff
@@ -29,6 +29,8 @@ protected slots:
 	void onThreadStarted();
 
 private:
+	QString hwType_;
```
Comment (dretrex) [inline source/AMDSThreadedTcpDataServer.h:32]: What is a hwType?  If it's an abbreviation, why are we abbreviating it?

File: source/AMDSThreadedTcpDataServer.h [hunk 3]
```diff
@@ -14,7 +14,7 @@ class AMDSThreadedTcpDataServer : public QObject
 	Q_OBJECT
 public:
 	/// Constructs a TCP server object, and passes it to a new thread
-	explicit AMDSThreadedTcpDataServer(QObject *parent = 0);
+	explicit AMDSThreadedTcpDataServer(QString hwType, QObject *parent = 0);
```
Comment (dretrex) [inline source/AMDSThreadedTcpDataServer.h:17]: This should also have been a `const QString &hwType`.

File: source/AMDSCentralServer.h [hunk 4]
```diff
@@ -15,7 +15,7 @@ class AMDSCentralServer : public QObject
 {
 Q_OBJECT
 public:
-	AMDSCentralServer(QObject *parent = 0);
+	AMDSCentralServer(QString hwType, QObject *parent = 0);
```
Comment (dretrex) [inline source/AMDSCentralServer.h:18]: This should also should be a `const QString &`.

File: source/AMDSThreadedTcpDataServer.h (hunk 2 above)
Comment (XuetaoLiu-S) [inline source/AMDSThreadedTcpDataServer.h:32]: this is removed in a later PR.

File: source/AMDSThreadedTcpDataServer.h (hunk 3 above)
Comment (XuetaoLiu-S) [inline source/AMDSThreadedTcpDataServer.h:17]: This is changed as well. Now, the network interface is passing via QSetting

**Pull request merged**